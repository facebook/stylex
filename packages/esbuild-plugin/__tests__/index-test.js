/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 *
 */

'use strict';

import path from 'path';
import esbuild from 'esbuild';
import stylexPlugin from '../src/index';

async function build(options = {}) {
  const { outputFiles } = await esbuild.build({
    entryPoints: [path.resolve(__dirname, '__fixtures__/index.js')],
    external: ['@stylexjs/stylex'],
    minify: false,
    bundle: true,
    write: false,
    plugins: [stylexPlugin({ useCSSLayers: true, ...options })],
  });

  return { js: outputFiles[0], css: outputFiles[1] };
}

describe('esbuild-plugin-stylex', () => {
  it('extracts and bundles CSS without inject calls, bundles JS', async () => {
    const { js, css } = await build();

    expect(js.text).toMatchSnapshot();

    expect(css.text).toMatchSnapshot();
  });

  it('preserves stylex.inject calls and does not extract CSS in development mode', async () => {
    const { js, css } = await build({ dev: true });

    expect(js.text).toMatchSnapshot();

    expect(css).toBeUndefined();
  });
});
