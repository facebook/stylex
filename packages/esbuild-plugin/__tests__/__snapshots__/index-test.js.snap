// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esbuild-plugin-stylex extracts and bundles CSS without inject calls, bundles JS 1`] = `
""use strict";
(() => {
  var __create = Object.create;
  var __defProp = Object.defineProperty;
  var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
  var __getOwnPropNames = Object.getOwnPropertyNames;
  var __getProtoOf = Object.getPrototypeOf;
  var __hasOwnProp = Object.prototype.hasOwnProperty;
  var __require = /* @__PURE__ */ ((x) => typeof require !== "undefined" ? require : typeof Proxy !== "undefined" ? new Proxy(x, {
    get: (a, b) => (typeof require !== "undefined" ? require : a)[b]
  }) : x)(function(x) {
    if (typeof require !== "undefined")
      return require.apply(this, arguments);
    throw Error('Dynamic require of "' + x + '" is not supported');
  });
  var __copyProps = (to, from, except, desc) => {
    if (from && typeof from === "object" || typeof from === "function") {
      for (let key of __getOwnPropNames(from))
        if (!__hasOwnProp.call(to, key) && key !== except)
          __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
    }
    return to;
  };
  var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
    // If the importer is in node compatibility mode or this is not an ESM
    // file that has been converted to a CommonJS file using a Babel-
    // compatible transform (i.e. "__esModule" has not been set), then set
    // "default" to the CommonJS "module.exports" for node compatibility.
    isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
    mod
  ));

  // __tests__/__fixtures__/index.js
  var import_stylex3 = __toESM(__require("@stylexjs/stylex"), 1);

  // __tests__/__fixtures__/fooStyles.js
  var import_stylex = __toESM(__require("@stylexjs/stylex"), 1);
  var fooStyles_default = {
    foo: {
      display: "xt0psk2",
      width: "xh8yej3",
      $$css: true
    }
  };

  // __tests__/__fixtures__/bazStyles.js
  var import_stylex2 = __toESM(__require("@stylexjs/stylex"), 1);
  var bazStyles_default = {
    baz: {
      display: "x1lliihq",
      height: "xnsd7bc",
      width: "x3hqpx7",
      $$css: true
    }
  };

  // __tests__/__fixtures__/index.js
  var styles = {
    bar: {
      animationName: "x127lhb5",
      display: "x78zum5",
      marginLeft: "x16ydxro",
      marginInlineStart: null,
      marginInlineEnd: null,
      height: "x1xa6b72",
      backgroundColor: "xrkmrrc x1r3o6fz",
      $$css: true
    }
  };
  function App() {
    return import_stylex3.default.props(fooStyles_default.foo, styles.bar, bazStyles_default.baz);
  }
})();
"
`;

exports[`esbuild-plugin-stylex extracts and bundles CSS without inject calls, bundles JS 2`] = `
"
@layer priority1, priority2, priority3;
@layer priority1{
@keyframes xekv6nw-B{0%{opacity:0;}100%{opacity:1;}}
}
@layer priority2{
.x127lhb5{animation-name:xekv6nw-B}
.xrkmrrc{background-color:red}
.x1lliihq{display:block}
.x78zum5{display:flex}
.xt0psk2{display:inline}
.x1r3o6fz:hover{background-color:pink}
}
@layer priority3{
.x1xa6b72{height:700px}
.xnsd7bc{height:900px}
.x16ydxro{margin-left:10px}
.xh8yej3{width:100%}
.x3hqpx7{width:50%}
}"
`;

exports[`esbuild-plugin-stylex preserves stylex.inject calls and does not extract CSS in development mode 1`] = `
""use strict";
(() => {
  var __create = Object.create;
  var __defProp = Object.defineProperty;
  var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
  var __getOwnPropNames = Object.getOwnPropertyNames;
  var __getProtoOf = Object.getPrototypeOf;
  var __hasOwnProp = Object.prototype.hasOwnProperty;
  var __require = /* @__PURE__ */ ((x) => typeof require !== "undefined" ? require : typeof Proxy !== "undefined" ? new Proxy(x, {
    get: (a, b) => (typeof require !== "undefined" ? require : a)[b]
  }) : x)(function(x) {
    if (typeof require !== "undefined")
      return require.apply(this, arguments);
    throw Error('Dynamic require of "' + x + '" is not supported');
  });
  var __copyProps = (to, from, except, desc) => {
    if (from && typeof from === "object" || typeof from === "function") {
      for (let key of __getOwnPropNames(from))
        if (!__hasOwnProp.call(to, key) && key !== except)
          __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
    }
    return to;
  };
  var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
    // If the importer is in node compatibility mode or this is not an ESM
    // file that has been converted to a CommonJS file using a Babel-
    // compatible transform (i.e. "__esModule" has not been set), then set
    // "default" to the CommonJS "module.exports" for node compatibility.
    isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
    mod
  ));

  // __tests__/__fixtures__/index.js
  var import_stylex_inject3 = __toESM(__require("@stylexjs/stylex/lib/stylex-inject"), 1);
  var import_stylex3 = __toESM(__require("@stylexjs/stylex"), 1);

  // __tests__/__fixtures__/fooStyles.js
  var import_stylex_inject = __toESM(__require("@stylexjs/stylex/lib/stylex-inject"), 1);
  var import_stylex = __toESM(__require("@stylexjs/stylex"), 1);
  var _inject2 = import_stylex_inject.default;
  _inject2(".xt0psk2{display:inline}", 3e3);
  _inject2(".xh8yej3{width:100%}", 4e3);
  var fooStyles_default = {
    foo: {
      fooStyles__foo: "fooStyles__foo",
      display: "xt0psk2",
      width: "xh8yej3",
      $$css: true
    }
  };

  // __tests__/__fixtures__/bazStyles.js
  var import_stylex_inject2 = __toESM(__require("@stylexjs/stylex/lib/stylex-inject"), 1);
  var import_stylex2 = __toESM(__require("@stylexjs/stylex"), 1);
  var _inject22 = import_stylex_inject2.default;
  _inject22(".x1lliihq{display:block}", 3e3);
  _inject22(".xnsd7bc{height:900px}", 4e3);
  _inject22(".x3hqpx7{width:50%}", 4e3);
  var bazStyles_default = {
    baz: {
      bazStyles__baz: "bazStyles__baz",
      display: "x1lliihq",
      height: "xnsd7bc",
      width: "x3hqpx7",
      $$css: true
    }
  };

  // __tests__/__fixtures__/index.js
  var _inject23 = import_stylex_inject3.default;
  _inject23("@keyframes xekv6nw-B{0%{opacity:0;}100%{opacity:1;}}", 1);
  _inject23(".x127lhb5{animation-name:xekv6nw-B}", 3e3);
  _inject23(".x78zum5{display:flex}", 3e3);
  _inject23(".x16ydxro{margin-left:10px}", 4e3);
  _inject23(".x1xa6b72{height:700px}", 4e3);
  _inject23(".xrkmrrc{background-color:red}", 3e3);
  _inject23(".x1r3o6fz:hover{background-color:pink}", 3130);
  var styles = {
    bar: {
      "index__styles.bar": "index__styles.bar",
      animationName: "x127lhb5",
      display: "x78zum5",
      marginLeft: "x16ydxro",
      marginInlineStart: null,
      marginInlineEnd: null,
      height: "x1xa6b72",
      backgroundColor: "xrkmrrc x1r3o6fz",
      $$css: true
    }
  };
  function App() {
    return import_stylex3.default.props(fooStyles_default.foo, styles.bar, bazStyles_default.baz);
  }
})();
"
`;
